Dashboard/ViewModels/ScrutineeringViewModel.cs

```diff
@@ -0,0 +1,134 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using Dashboard.Models;
+using YamlDotNet.Serialization;
+using YamlDotNet.Serialization.NamingConventions;
+
+namespace Dashboard.ViewModels;
+
+public class ScrutineeringViewModel : ViewModelBase, IDisposable
+{
+    private readonly IDataStore _dataStore;
+    private readonly FileSystemWatcher _fileWatcher;
+    private YamlData _yamlData = new();
+
+    public ScrutineeringViewModel(IDataStore dataStore)
+    {
+        // This constructor is used for design-time data, so we don't need to start the connector
+        _dataStore = dataStore;
+        _dataStore.DataUpdated += OnDataChanged;
+
+        // The folder we are in at runtime is net8.0 (AV-Datalogger/Dashboard/bin/Debug/net8.0/Dashboard.exe), as
+        // our yaml file is in the resources folder, exit the current folder three times
+        var pathDirectory = "../../../Resources/";
+
+        // // Initialize file watcher
+        _fileWatcher = new FileSystemWatcher
+        {
+            // Watch the project directory
+            Path = pathDirectory,
+            // Watch specifically for yaml files
+            Filter = "*.yaml",
+            // Trigger on file modifications
+            NotifyFilter = NotifyFilters.LastWrite
+        };
+
+        // Link the event handler to react when the file changes
+        _fileWatcher.Changed += OnFileChanged;
+
+        // Setting this to true starts monitoring the file for changes.
+        _fileWatcher.EnableRaisingEvents = true;
+
+        // Load the initial YAML data when the ViewModel is created
+        LoadYamlData(Path.Combine(pathDirectory, "AV_Inspection_Flow.yaml"));
+    }
+
+    public double SteeringAngle => _dataStore.SteeringAngle;
+
+    /// <summary>
+    ///     Getter and Setter for the flow of steps for autonomous vehicle inspection
+    /// </summary>
+    public YamlData YamlData
```

Comment:
if u add the [ObservableProperty] annotation, it will do this get set magic for free

Reply:
would it still call OnPropertyChanged?

Reply:
4356d1ba7cd3de49f2f83f3bd5b5cf2850fb9488